client/src/app/categories-page/categories-page.component.ts

```diff
@@ -0,0 +1,145 @@
+  isWarning: boolean = false;
+
+  addCategoryForm = this.fb.group({
+    categoryType: ['expense'],
+    title: ['', [Validators.required]]
+  })
+
+  constructor(
+    private fb: FormBuilder,
+    private categoryService: CategoriesService
+  ) { }
+
+  ngOnInit(): void {
+    this.getCategories();
+  }
+
+  getCategories(type: any = null, searchedText: string = '') {
+    this.categoryService
+      .getCategories(type)
+      .pipe(untilDestroyed(this))
+      .subscribe({
+        next: data => {
+          this.categories = data.categories
+          this.categories = this.categories.filter(category => category.title.includes(searchedText))
+        }
+      })
+  }
+
+  addCategory() {
+    if (this.addCategoryForm.valid) {
+      const { categoryType, title } = this.addCategoryForm.value;
+      this.categoryService
+        .addCategory(categoryType, title)
+        .pipe(untilDestroyed(this))
+        .subscribe({
+          next: data => {
+            this.toasterMessageStatus = true;
+            this.toasterMessage = data.message;
+            this.isOpenAddCategoryModal = false;
+            setTimeout(() => {
+              this.toasterMessageStatus = false;
+              window.location.reload()
+            }, 2500)
+          },
+          error: error => {
+            this.isAddCategoryError = true;
+            this.addCategoryErrorMessage = error.error.message;
+          }
+        })
+    }
+  }
+
+  filterCategory(type: string) {
+    this.getCategories(type);
+    if (type === 'income') {
+      this.isActiveIncomeCategories = !this.isActiveIncomeCategories;
+      if (!this.isActiveIncomeCategories) this.getCategories();
+      if (this.isActiveExpenseCategories) this.isActiveExpenseCategories = false;
+    }
+    if (type === 'expense') {
+      this.isActiveExpenseCategories = !this.isActiveExpenseCategories
+      if (!this.isActiveExpenseCategories) this.getCategories();
+      if (this.isActiveIncomeCategories) this.isActiveIncomeCategories = false;
+    }
+  }
+
+  closeAddCategoryModal() {
+    this.isOpenAddCategoryModal = false;
+  }
+
+  closeVerifyModal() {
+    this.verifyModalStatus = false;
+  }
+
+  openAddCategoryModal() {
+    this.isOpenAddCategoryModal = true;
+  }
+
+  openVerifyModal(categoryId: number, title: string, message: string) {
+    this.verifyModalStatus = true;
+    this.verifyTitle = title;
+    this.verifyMessage = message;
+    this.deletedCategoryId = categoryId;
+  }
+
+  searchCategory(searchedText: string) {
+    this.getCategories(null, searchedText)
+  }
+
+  deleteCategory() {
+    this.categoryService
+      .deleteCategory(this.deletedCategoryId)
+      .pipe(untilDestroyed(this))
+      .subscribe({
+        next: data => {
+          this.isWarning = false;
+          this.toasterMessage = data.message;
+          this.toasterMessageStatus = true;
+          setTimeout(() => {
+            this.toasterMessageStatus = false;
+            window.location.reload();
```

Comment:
Don't reload

Reply:
I will use route.navigateByUrl('')

Reply:
I don't understand why you want to navigate